### ISSUE TITLE
Help Needed: Enhance SEO and Analytics for Better Search Engine Ranking

### ISSUE THREAD

**Issue by yada** [microcks/microcks.io#209]
### Reason/Context

Hi Microcks Community,

Despite having a well-designed website, we're struggling to rank high on search engines for key terms related to our project. To improve our visibility and reach, we need help and contributions from experts in SEO, sitelinks, and Google Analytics.

### Description

How You Can Help:
We would greatly appreciate your contributions if you have expertise in these areas.
Please share any insights, tools, or resources that could assist us in improving our search engine ranking.
Feel free to fork the repository, make changes, and submit a pull request with your improvements.

Thank you in advance for your support and contributions!

Best regards,
The Microcks Team

### Implementation ideas

Areas of Focus:

SEO Optimization:
- Improve on-page SEO elements (meta tags, keywords, content optimization).
- Enhance off-page SEO strategies (backlinks, social sharing).
- Conduct keyword research to identify high-impact keywords.
- Optimize site speed and mobile responsiveness.

Sitelinks:
- Ensure our website structure supports sitelinks.
- Implement necessary schema markup.
- Optimize internal linking to highlight important pages.

Google Analytics:
- Update existing Google Analytics to track key metrics.
- Implement goal tracking for conversions.
- Analyze user behavior to identify areas for improvement.
- Provide regular reports on website performance.

And anything else that can help 🙏

**Comment by yada** [comment]
Some cleanup has been done on Google Search Console to delete all (around 100 URLs !) 404 old URLs (documentation reorg...). This should help in the coming weeks/months -> to follow.

**Comment by yada** [comment]
There are still many URLs to fix. Google indexation is taking a long time.
Adding a new list of URLs to delete from the index, to follow.

**Comment by yada** [comment]
We're moving slowly but surely regarding Google indexation. "90 pages of your site have been validated as corrected."

**Comment by divyansh-prks** [comment]
hey @yada  for this issue my approach would be like this - 

For seo optimization -> 
1. I will research about Trending and popular keyboards related to our website and put it into our meta tags 
2. For the faster rendering of the website i will use lazy loading ,  compress image  , Backlinks . 
3. I will post about the website on different platform and post blogs about it too . (Twitter , youtube , medium , hashnode , linkedin etc )
4. I will research about the similar product or competitor with our website and find low ranking keyword so that it can improve SEO . 
5. Remove low-quality content
6. I will make sure that  Every piece of content on your website must have a purpose

Let me know if i can start working on this issue ?

**Comment by yada** [comment]
> Please creat hight authortys backlinks for your websit

Hi @samikhan143 ,
I am very sorry for the late reply... (missed notifications...)
Thanks for the info; I was not aware of this possibility; is there any pointer or tool to do it?

**Comment by yada** [comment]
Today, we still have 107 pages with a 404 error 🥵 This is due to the significant documentation refactoring done a few months ago and the fact we have not done any backlinks redirection... we expected Google to notice and remove them... but not...  

So, do we need to create backlinks for all of them (point to the new URLs)?

Here is the list of URLs concerned:
[urls.txt](https://github.com/user-attachments/files/18759438/urls.txt)

**Comment by divyansh-prks** [comment]
let me research for this i will try to provide efficient solution as soon as possible

**Comment by yada** [comment]
I'll ask (google search) to delete (do not index) these URLs:
```
https://microcks.io/images/artifacts-upload_hu3c9aa59d797e441dd74f9c3c3ec7292f_75005_1216x676_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/deployment-partially-managed_hu693445f18b5fe4f3adc22011e3bfecba_452303_2966x1724_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/postman-export_hu26667b48f922f16e766003f60efbf420_68630_986x566_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/service-account-clients_hu2a249c101e358044bf4e8322196559b5_368293_3208x1674_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/architecture-simple_hu95d85a2390c04a02a4ae6848acdf75c1_336534_2830x1582_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/parameters-constraints_huf5aa20a418af9f7d2fff57ffe1def3da_245311_2642x762_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/asyncapi-mocks_hu81fadcbcc6c8e9542289a42f759b2ce2_304929_2688x1810_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/importer-step1_hu9067fd24a78ae1bdee3f898df844e2c2_171208_1816x1238_resize_q90_h2_lanczos_3.webp
https://microcks.io/images/postman-share-api-link_hu45a91574019d5d819ac08141900872c3_112623_1577x733_resize_q90_h2_lanczos_3.webp
```

After a quick check, all the others need backlinks ex:
from https://microcks.io/documentation/using/templates/ 🛑
to https://microcks.io/documentation/references/templates/ ✅

**Comment by yada** [comment]
I have asked Google Analytics for a re-indexation (ongoing), as it seems that it has not been done for a while... so let's see if it removes (or not) the 404 (from my understanding, it should)... to follow.

**Comment by yada** [comment]
@divyansh-prks Still interested in contributing to this issue to optimize Microcks website?

**Comment by divyansh-prks** [comment]
yes @yada  i am still interested for contributing in microcks website . I was busy in my exams that's why couldn't able to work more from last couple of days

**Comment by yada** [comment]
> yes [@yada](https://github.com/yada) i am still interested for contributing in microcks website . I was busy in my exams that's why couldn't able to work more from last couple of days

Thanks, have good exams and all the best ;) 
Let's catch up on this when you are available. will keep it open.

**Comment by Vaishnav88sk** [comment]
Prs: #536 , #537

**Comment by Vaishnav88sk** [comment]
### Status Update

We have recently made some progress on the codebase to address the SEO and Sitelinks aspects of this issue:
- **Schema Markup & Sitelinks:** Implemented JSON-LD structured data (`Article`, `BlogPosting`, `TechArticle`) and canonical URLs to help search engines better understand the site structure. #536 
- **On-page SEO:** Added a `robots.txt` file and updated page metadata/titles for better bot management and indexing. #535 

**What remains to be tackled (from a code perspective):**
1. **Analytics Goal Tracking:** We need to add custom Google Analytics event triggers (e.g., `gtag('event', 'click'...)`) to our primary CTA buttons (like downloads or Discord links) to track conversions.
2. **Site Speed & Mobile Responsiveness:** Images and iframes still need to be optimized for lazy loading (e.g., adding `loading="lazy"` attributes).
3. **Internal Linking:** Automating breadcrumbs for docs and adding dynamic "Related Posts" sections at the bottom of blogs to strengthen internal navigation.
